aws-cli/bash-linux/s3/bucket-lifecycle-operations/bucket-operations-test.sh

```diff
@@ -29,61 +21,72 @@ function usage {
     echo ""
     echo "To pause the script between steps so you can see the results in the"
     echo "AWS Management Console, include the parameter -i."
+    echo ""
+    echo "IMPORTANT: Running this script creates resources in your Amazon"
+    echo "   account that can incur charges. It is your responsibility to"
+    echo "   ensure that no resources are left in your account after the script"
+    echo "   completes. If an error occurs during the operation of the script,"
+    echo "   then resources can remain that you might need to delete manaully."
 }
 
 # Parse parameters
-    if [ $# -gt 1 ]
-    then
-        echo "Too many parameters."
-        usage
-        exit 1
-    fi
-
-    if [[ $# == 0 ]]; then
+    if [ "$#" -eq "0" ]; then 
         interactive=false
-    elif [[ "$1" == "-i" ]]; then
+    elif [ "$1" == "-i" ] && [ "$#" -eq "1" ]; then
         interactive=true
     else
         echo "Invalid parameter."
         usage
         exit 1
-   fi
-
-# Initialize the shell scripts RANDOM variable
-RANDOM=$$
-# Configure random number generator to issue numbers between 1000 and 9999, inclusive
-DIFF=$((9999-1000+1))
-# Generate 3 random numbers for the bucket name
-R1=$(($(($RANDOM%$DIFF))+X))
-R2=$(($(($RANDOM%$DIFF))+X))
-R3=$(($(($RANDOM%$DIFF))+X))
-
-bucketname="testbucket-$R1-$R2-$R3"
-filename="testfile.txt"
+    fi
+
+bucketname=$(generate-random-name s3test)
+filename1=$(generate-random-name s3testfile)
+filename2=$(generate-random-name s3testfile)
 region="us-west-2"
 
-echo "Starting tests of bucket operations"
- 
+iecho "bucketname=$bucketname"
+iecho "filename1=$filename1"
+iecho "filename2=$filename2"
+iecho "region=$region"
+
+iecho "Starting tests of bucket operations"
+
+# The functions we want to test all come from this file 
```

Comment:
All comments should be grammatically correct and complete sentences, including punctuation.

Reply:
Fixed.